### ISSUE TITLE
Allow procedural textures to override their sampler state

### ISSUE THREAD

**Issue by rileyzzz** [Facepunch/sbox-issues#3010]
### What it is?

Right now, it seems like overriding a texture or setting a texture attribute on a material will automatically use the sampler state of the associated texture input in the shader.
However, in situations where I've got a common shader used by multiple objects, I really don't want to have different materials and shader features for each combination of addressing mode, filtering, anisotropy level, etc. Specifically, I need to be able to set the S, T, and U wrapmodes separately at runtime.

### What should it be?

Looking at the disassembly of TextureBuilder, it seems like it already supports flags like TSPEC_SUGGEST_CLAMPS/T/U, TSPEC_LINEAR_COLOR_SPACE etc. Not sure if these actually work in the engine, but if they do, it'd be cool if we had a few extra functions on TextureBuilder that let us control them.
Alternatively, it'd be cool if we had something like an optional "WithSamplerStateOverride" on TextureBuilder that let us override the complete sampler state, including whether textures clamp/wrap/mirror, how they're filtered, whether they're given an sRGB image format in dx11 or not, the max anisotropy level, etc.

For accurate runtime importers/parsers, setting this stuff per-texture is a must, since in many cases compiled texture formats will include their associated values for wrapping modes, filtering, etc alongside the texture data.

**Comment by aylaylay** [comment]
Yeah I hate this too, I wish you could set sampler state on the texture instead of shader but the engine doesn't work that way and it would be a headache to change it over last time I checked. I'm sure it will get solved one day, I just don't know what day it will be.
